sionna/channel/optical/fiber.py

```diff
@@ -0,0 +1,309 @@
+    symmetrized SSFM according to Eq. (7) of [SymSSFM]_ and can be written as
+
+    .. math::
+
+        E(z+\Delta_z,t) \approx \exp\left(\frac{\Delta_z}{2}\hat{D}\right)\exp\left(\int^{z+\Delta_z}_z \hat{N}(z')dz'\right)\exp\left(\frac{\Delta_z}{2}\hat{D}\right)E(z,\,t)
+
+    The integral is approximated by :math:`\Delta_z\hat{N}`.
+
+    Additionally, ideally distributed Raman amplification may be applied, which
+    is implemented as in [RamanASE]_.
+
+    The SSFM operates on normalized time :math:`T_\text{norm}` (e.g., 1 ps) and
+    distance units :math:`L_\text{norm}` (e.g., 1 km). Hence, parameters as well
+    as the signal itself have to be given in the same normalized units. Note
+    that, despite the normalization, the SSFM is implemented with physical units
+    different from the normalization, e.g., used for the nonlinear
+    Fourier transform.
+
+    This class inherits from the Keras `Layer` class and can be used as layer in
+    a Keras model.
+
+    Example
+    --------
+
+    Setting-up:
+
+    >>> ssfm_channel = SSFM(
+    >>>     0.046, -21.67, 193.55e12, 1.27, 100, 80, 200, 1e-17, 1.0,
+    >>>     True, True, True, True)
+
+    Running:
+
+    >>> y = ssfm_channel([1.0+1.0j, 1.0+1.0j, 1.0+1.0j])
+
+    Parameters
+    ----------
+        alpha : float
+            Attenuation coefficient :math:`\alpha` in :math:`(1/L_\text{norm})`
+
+        beta_2 : float
+            Chromatic dispersion coefficient :math:`\beta_2` in :math:`(T_\text{norm}^2/L_\text{norm})`
+
+        f_c : float
+            Carrier frequency :math:`f_\mathrm{c}` in :math:`(1/T_\text{norm})`
+
+        gamma : float
+            Kerr-nonlinearity :math:`\gamma` in :math:`(1/L_\text{norm}/W)`
+
+        half_window_length : int
+            Half of the Hammwing window length
+
+        ell : float
+            Fiber length :math:`\ell` in :math:`(L_\text{norm})`
+
+        n_ssfm : int
+            Number of steps :math:`N_\mathrm{SSFM}`
+
+        n_sp : float
+            Spontaneous emission factor :math:`n_\mathrm{sp}` of Raman amplification
+
+        dt : float
+            Normalized time step :math:`\Delta_t` in :math:`(T_\text{norm})`
+
+        t_norm : float
+            Time normalization :math:`T_\text{norm}` in :math:`(s)`.
+
+        with_attenuation : bool
+            Enables application of attenuation
+
+        with_amplification : bool
+            Enables application of ideal inline amplification and corresponding
+            noise
+
+        with_dispersion : bool
+            Enables application of chromatic dispersion
+
+        with_nonlinearity : bool
+            Enables application of Kerr nonlinearity
+
+        swap_memory : bool
+            Use CPU memory for while loop. Defaults to True.
+
+        dtype : Complex tf.DType
+            Defines the datatype for internal calculations and the output
+            dtype. Defaults to `tf.complex64`.
+
+    Input
+    -----
+        x : Tensor, tf.complex
```

Comment:
I have removed the input tuple as it did not seem to be used.

Reply:
OK